app/policies/rent_policy.rb

```diff
@@ -0,0 +1,10 @@
+class RentPolicy < ApplicationPolicy
+  def index?
+    record.first.user_id == user.id
```

Comment:
Shouldn't all records have the same user_id?
And also, what you are trying to compare is the user_id you receive from the params vs the one who is logged in. Wouldn't it be better to compare that instead of the user in the models you got from the query? The query should be tested and only bring models from the user who's param was received.